data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/index/DocumentBuilder.java

```diff
@@ -2,16 +2,25 @@
 
 import org.opensearch.dataprepper.model.event.Event;
 
+import java.util.List;
+
 public final class DocumentBuilder {
 
-    public static String build(final Event event, final String documentRootKey, final String tagsTargetKey) {
-        if (documentRootKey != null && event.containsKey(documentRootKey)) {
-            final String document = event.getAsJsonString(documentRootKey);
-            if (document == null || !document.startsWith("{")) {
-                return String.format("{\"data\": %s}", document);
-            }
-            return document;
+    public static String build(final Event event, final String documentRootKey, final String tagsTargetKey, final List<String> includeKeys, final List<String> excludeKeys) {
+        final String document = event.jsonBuilder()
+                .rootKey(documentRootKey)
+                .includeKeys(includeKeys)
+                .excludeKeys(excludeKeys)
+                .includeTags(tagsTargetKey)
+                .toJsonString();
+
+        if (document == null || !document.startsWith("{")) {
+            return String.format("{\"data\": %s}", document);
```

Comment:
You didn't introduce this, but `String.format` is slow. String concatenation would be faster.

Feel free to update that, but since you didn't introduce it, no need to make the change to accept the PR.